sdk/go/auto/remote_workspace_test.go

```diff
@@ -130,9 +153,16 @@ func testRemoteStackGitSource(t *testing.T, fn func(ctx context.Context, stackNa
 	stackName := FullyQualifiedStackName(pulumiOrg, pName, sName)
 	repo := GitRepo{
 		URL:         remoteTestRepo,
-		Branch:      "refs/heads/master",
 		ProjectPath: "goproj",
 	}
+	if useCommitHash {
+		commitHash, err := fetchCommitHash(remoteTestRepo, remoteTestRepoBranch)
+		require.NoError(t, err)
+		require.Regexp(t, "^[0-9a-f]{40}$", commitHash)
```

Comment:
I would treat the commit hash as an opaque string here.
Git is looking to move to SHA256 long term and the length of hashes may change.

```suggestion
```

This is unlikely to be a problem anywhere in the near term, but this extra check seems unnecessary unless I'm missing something about what this change is hoping to verify. If we want to keep the check, maybe it can be 40-64:

```suggestion
		require.Regexp(t, "^[0-9a-f]{40,64}$", commitHash)
```